test/bin/build_images.sh

```diff
@@ -288,10 +290,16 @@ if [ -z "${release_info_rpm}" ]; then
     error "Failed to find microshift-release-info RPM in ${LOCAL_REPO}"
     exit 1
 fi
+release_info_rpm_main=$(find "${MAIN_REPO}" -name 'microshift-release-info-*.rpm' | sort | tail -n 1)
+if [ -z "${release_info_rpm_main}" ]; then
+    error "Failed to find microshift-release-info RPM in ${MAIN_REPO}"
+    exit 1
+fi
 SOURCE_VERSION=$(rpm -q --queryformat '%{version}' "${release_info_rpm}")
 MINOR_VERSION=$(echo "${SOURCE_VERSION}" | cut -f2 -d.)
 PREVIOUS_MINOR_VERSION=$(( "${MINOR_VERSION}" - 1 ))
 FAKE_NEXT_MINOR_VERSION=$(( "${MINOR_VERSION}" + 1 ))
+SOURCE_VERSION_MAIN=$(rpm -q --queryformat '%{version}' "${release_info_rpm_main}")
```

Comment:
The `release-4.Y -> pr-for-release-4.Y` test wouldn't occur until after the code is merged into main an cherry-picked to the release branch, right?  That would still be a useful test to have I think, but wouldn't provide signal prior to merging the PR.  

Also, I think in cases where the cherrypick is not a backport to older branches, testing that release-4.14 -> PR-for-release-4.14 is functionally the same as main -> PR because the current branches are kept up to date with main.

Reply:
> This made me think, do we need `main` -> `PR` or `release-4.1*` -> `PR`? @dhellmann WDYT?

We want to test updating from the previous branch and within the current branch.

We have a test that installs the latest Y-1 package and upgrades to the PR already via https://github.com/openshift/microshift/blob/main/test/scenarios/upgrade-minor-version-healthy.sh.

We could add a test that builds the HEAD of the Y-1 branch and upgrades to the current PR, too, although we've seen issues trying to use different versions of go to compile different branches so I'm not sure if the effort would be worth the value. Let's do that separately, if we do it.

> When we get to 4.15, we probably want presubmits for 4.14 run from `release-4.14`, right? Not `main/release-4.15 -> PR-for-4.14`, but `release-4.14 -> PR-for-4.14`

This is a good point. We need to clone the right branch as part of this new scenario. It should be possible to use the version information in the Makefile to build the `release-$(MAJOR).$(MINOR)` branch name and clone that instead of `main`.

As part of that change, we could call this "upgrade from HEAD" instead of "upgrade from main", and make all of the relevant file and variable name changes.

> The `release-4.Y -> pr-for-release-4.Y` test wouldn't occur until after the code is merged into main an cherry-picked to the release branch, right? That would still be a useful test to have I think, but wouldn't provide signal prior to merging the PR.

A PR submitted against `release-4.y` should be tested by upgrading from HEAD of `release-4.y`.

> Also, I think in cases where the cherrypick is not a backport to older branches, testing that release-4.14 -> PR-for-release-4.14 is functionally the same as main -> PR because the current branches are kept up to date with main.

This is true. The version in our main branch will not change until the first rebase that updates to a nightly version of OCP with the new version. We will have to see how long that period is to understand what impact it will have on this test.

Reply:
Looking at [rebase-test log](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_microshift/2144/pull-ci-openshift-microshift-main-test-rebase/1685410855337857024/artifacts/test-rebase/openshift-microshift-rebase/build-log.txt) it looks like we can get info about branch to checkout from envs:
```
PULL_BASE_REF=main
```